tsc/webusb/index.ts

```diff
@@ -257,32 +244,39 @@ export class WebUSB implements USB {
         // Pre-filter devices
         devices = this.preFilterDevices(devices, preFilters);
 
-        const webDevices: USBDevice[] = [];
+        const refreshedKnownDevices = new Map<usb.Device, WebUSBDevice>();
+
+        await Promise.all(devices.map(async (device) => {
+            const webDevice = await this.getWebDevice(device);
+
+            if (webDevice) {
+                refreshedKnownDevices.set(device, webDevice);
+            }
+        }));
```

Comment:
```suggestion
        devices.forEach(async device => {
            const webDevice = await this.getWebDevice(device);

            if (webDevice) {
                refreshedKnownDevices.set(device, webDevice);
            }
        });
```

Reply:
We'd want to await for all the `getWebDevice` to complete before going ahead, no? Using `forEach` might not do that.

Reply:
Your right, it need to be a normal loop...

Reply:
Suggestion updated

Reply:
Committed.